### ISSUE TITLE
Opentoonz deleting/moving frames

### ISSUE THREAD

**Issue by Randomwords247** [opentoonz/opentoonz#5087]
### Description

Hi, I've had this happen a few times over the last few months with opentoonz. I've been working on an animation/animatic here and there and sometimes when I reopen my saved file, it'll turn out that a bunch of frames have either been moved, or have been deleted.

Last night I was working on this animation and finished off to go to bed. I saved everything, I even exported the file and it came out fine. I tend to draw some things that may overlap on different layers, so I have typically 2-3 layers I'm working on with people and 1 layer for backgrounds. These two frames had two layers used each, and this is how they looked last night in the rendered progress I had made
![image](https://github.com/opentoonz/opentoonz/assets/144225170/79dd57f6-5844-48da-802a-fe6ff942ef7c)
This is the frames now
![image](https://github.com/opentoonz/opentoonz/assets/144225170/5e2362fd-5706-4bec-8b1d-f7815be8fc4b)
![image](https://github.com/opentoonz/opentoonz/assets/144225170/1f1eb14b-e365-4ce7-8cc1-3cbb971927da)

The frames show a red dot icon and this is not the first time this has happened to me. This time it was only two frames, but I've had it before where I've opened up my file only to see multiple frames gone, and some straight up shifted around in the place of the deleted frames, almost as if they absorbed the frames and replaced them. One time some frames had straight up merged and I had to detangle, so to speak, the frames from one another so I could put them back in the right spots without looking like I'd erased a bunch.

Why is this happening and how can I stop it? It's really infuriating to be working on something only to have half your work deleted. It always is upon reopening the file after saving everything, so I don't understand how in the world this could happen. It was all fine last night when I saved it, heck I was even able to render it and it was all fine, but now this morning its all changed

Please help. I don't want to keep losing my progress like this. Thankfully because I rendered it I can salvage the frames to an extent, but that's not something I want to rely on and it sucks that I have to

### Steps to Reproduce

1. I really have no idea. My file is currently almost 300 frames long, frame rate of 12fps, and has four layers in use (3 for characters one for background). As background information. They are all raster layers
2. It happens when I open up opentoonz and reopen the file. When I save everything, the frames are all in place, nothing has been shifted or deleted, everythings fine, but when I reopen the saved file it's like someone messed with it in the night
3. I have not messed with the files. I don't know what could be causing this.
4. This was happening on 1.7. I recently (the other week) switched to the nightly build in hopes it would've been patched or woudn't happen again. I'm currently on 1.7.1, but I need to emphasise THIS HAPPENED BEFORE I WAS ON THE NIGHTLY BUILD. I hoped the version change would fix it but it has Not.

### Expected Behavior

I expect opentoonz to not delete my frames?? That feels like a given. Like, please do not delete my hard work after I've already saved it when I open up the file. Reloading doesn't do anything btw, its permanantly stuck like that

### Screenshots, Video & Crash Logs

![image](https://github.com/opentoonz/opentoonz/assets/144225170/79dd57f6-5844-48da-802a-fe6ff942ef7c)
![image](https://github.com/opentoonz/opentoonz/assets/144225170/5e2362fd-5706-4bec-8b1d-f7815be8fc4b)
![image](https://github.com/opentoonz/opentoonz/assets/144225170/1f1eb14b-e365-4ce7-8cc1-3cbb971927da)

### OpenToonz Version

Nightly (Specify below)

### OpenToonz Version Information

1.7.1 (but also happened on 1.7)

### Operating System

Windows

### GPU

NVIDIA

### Graphics Tablet

Huion

**Comment by RodneyBaker** [comment]
Sorry to hear of the problems you are having @Randomwords247 

I have some recommendation but want to keep this as short as possible as I want to focus on preventing this from happening again in the future.  If the following sounds harsh... definitely not meant to be... I'm just trying to be brief and direct.

You said:

> I have not messed with the files. I don't know what could be causing this.

You have messed with the files.
That's what we do as artists and computer users.
We mess with files all the time.

> (when you return) frames have either been moved, or have been deleted.

Probably not deleted and very likely moved.
It is of course possible they have been deleted in the sense of being overwritten but... then the newly drawn files would be there and not the red (broken link) indicator.

99 times out of 100 (perhaps 99.8?) the red frame indicator is the result of files being moved.

Now... here is our task; we (or more appropriately you, want to know definitively how files on your computer might have been inadvertently moved.

Short aside:  The number one reason users of Opentoonz lose drawings/levels is because they save their scene but not their levels.  This happens when they use Save Scene thinking this will save their levels/drawing but it doesn't. 
Instead, very importantly, use Save All because we want to save all of our work... everything.
The Tahoma2D fork thought this important enough they have Save All as the only Save option in the main menu.
To get to the other save option the user has to drill down to find them.
I personally think they took that a step too far as I use Save As followed by a Save All quite often... but that's just me.

You mention that you exported but didn't give any details into what steps you use/used to export.
This is an area of supreme interest.
If you used the File > Export > Export Scene process then all of the 'currently saved' files of your current scene are in that newly created directory.  (I think OT will force a save if you haven't... will have to verify that however)

A decision you might have to make that the point of exporting in this way is whether to open that newly exported scene or continue using the one you are currently.  If we continue in the current scene then the files of that exported scene (in the new project directory) are now behind the times and won't have new frames and such that we create.  Unless/until we export again that is...

But exporting in this way is generally best for sharing our projects and capturing them at a moment in time (much like a backup).  A major plus of Exporting in this way is that all the files of our scene are included in that project directory AT THE TIME OF EXPORT.  So if you are missing files... that might be a great place to find them.

Our current scene is all important and the files in it we want to maintain until such a time that we set it aside in favor of one we deem to be our new (master) scene which we will then carry forward.

You mention that in times past your sequences got mixed up.
This often can be attributed to same name levels.  
For instance we might work in one project with level names A.tlv, B.pli, shotA.0001.png, etc.
Later we might work in another project and OT automatically generates A.tlv, B.pli... and we might name our image sequences created in another program 'shotA.0001.png'.   We might think that's all good because those files are in different directories.
But... if the files are the same name there is always a chance we might accidentally replace one with another of same name.
As such, caution should be used when using same name levels.
Again, this does not seem to be your issue as the red (missing level) indicator suggests the files have been moved/re-moved.

A question we might ask at this point is:  Where are your levels being stored/saved?

More deeply...
Since you are obviously working on a solid production and not just doodling about...
Are you creating a Project and then working in that project each time you open Opentoonz?


I'll stop there but there is much more to say and I definitely want to thank you for all of the information you have provided.
That helps immensely in understanding what you are experiencing and I plan to re-read your post again as you have provided lots of useful information that I surely have glossed over.

Let's get things going solidly for you  and where there are bugs/processes causing problem let's get those resolved too.

Hang in there!

**Comment by RodneyBaker** [comment]
Adding here in separate post a recomendation.
Right Click on your exposed cell in Timeline/Xsheet and open Level Settings.
Note where OT thinks the file should be and check that location on your harddrive.
You will probably have a .bak file there that is a renamed copy created at the last time the level was saved.
Removing the .bak extension (returning it to something like 'image.tlv' will let you know if it is (or is not) the correct file.

If the location being pointed to has anything strange to say... such as a location you aren't expecting... that's an important clue also.  I don't suspect that is your case however as you appear to be savy about file/project management.

**Comment by RodneyBaker** [comment]
Thahanks for the clarification.  Much appreciated.

You didn't come off as rude at all.
Frustrated... yes... and for good reason.

Yes, the title (or level name) isn't the same as the filename.
There are some good reasons for that although requests have been put in to have an option for those to be the same (or at least in sync).

As you've seen, it is an excellent practice to name the level when creating it as that will help to keep it unique and lessen the chance of one level getting confused with another with that exact same name.  Keep in mind here that we might want to use that level in multiple scenes... or even different projects... but it can help if the level name is "Heroguy.tlv" versus "A.tlv".
That is best done when creating the level although I have been in the habit recently of using Save Level As and then naming it as appropriate because... I am in the habit of creating new levels on the fly just by drawing into a new xsheet cell**.  By design that doesn't give us the prompt to create a new level it just creates it and lets us draw.

Consider however if you are just exposing levels according to a paper xsheet... A... C... GG... F... A... etc.
Thousands of exposures and very little time to do it.
Every keypress counts in that production environment.

Our production isn't usually as restricted and can slow down and enjoy the process.  ;)


There are many who have openly stated they hate the 'Sandbox' and there are good reasons for that.
I personally love the sandbox but I tend to do very short projects and so don't often start by creating a dedicated project to work in.  That is recommended for cases like yours.  In this way your files will all be in that project directory and not in the Sandbox.

There is a new option in Preferences in current nightlies that allows the column to be titled the same as the (first) level contained in that column.  That can be well worth turning on.  One downside to that approach:  Let's say we have all the parts of our character inside a toonz raster level named 'cybergirl.tlv'.  If we move the parts/frames of the level to different columns to layer out our character... all of the column headers will be named 'cybergirl'.  Obviously we'd be better suited to turn that option off and manually name the columns things like body, head, arms, big freakin weapon, etc.  Maybe not the weapon... that'd be better captured in its own (prop) level.  ;)


**I'll often create a new frame by tapping just off canvas with my pen which doesn't draw on the canvas but creates a new frame to draw in.  This works great with raster and toonz raster levels which are limited to drawing inside the raster bounding box... not so great for vector levels where that tap on the screen creates a little vector dot that will need to be removed later.


 Apologies for the verbosity.
I do want you to know there is light at the end of the tunnel and we (collectively) are very interested in improving workflows in Opentoonz.

The bottom line:  That red frame indicator is there because a link to a resource is broken.
We might not know how that breakage occurred but we want to be on the lookout for how to avoid it.

**Comment by RodneyBaker** [comment]
Will close as the original issue reported was likely resolved.
